response.go

```diff
@@ -28,18 +29,36 @@ type Response struct {
 	websocket *websocket.Conn
 	rtt       *time.Duration
 
-	content      []byte
-	contentState contentState
+	content       []byte
+	contentState  contentState
+	contentMethod string
 
 	cookies []*http.Cookie
 }
 
+type ErrorReader struct {
+	err error
+}
+
+func (er *ErrorReader) Read(_ []byte) (int, error) {
+	return 0, er.err
+}
+
+func (er *ErrorReader) Close() error {
+	return er.err
+}
+
+func newErrorReader(err error) *ErrorReader {
+	return &ErrorReader{err}
+}
+
 type contentState int
 
 const (
 	contentPending contentState = iota
 	contentRetreived
 	contentFailed
+	contentHijacked
```

Comment:
Could you please add comments to states? It's now plenty of them and they can be confusing.

* pending - we didn't try to retrieve response content
* retrieved - we retrieved it
* failed - we tried and failed
* hijacked - we transferred body reader to user and can't use